### PR TITLE
CDAP-19591 Source plugin configuration to accept GCS bucket location for datastream result

### PR DESCRIPTION
**Pull request by sumitjnn**
Enable user to specify the location where GCS bucket for Datastream results will be created. This is similar to how the GCS bucket location for staging bucket is accepted in https://github.com/data-integrations/bigquery-delta-plugins/

### CODE REVIEW

File: src/main/java/io/cdap/delta/datastream/DatastreamDeltaSource.java [hunk 1]
```diff
@@ -151,10 +151,10 @@ private void createStreamIfNotExisted(DeltaSourceContext context) throws IOExcep
       // check whether GCS Bucket exists first
       String bucketName = config.getGcsBucket();
       // If user doesn't provide bucketName, we assign one based on run id
-      if (bucketName == null) {
+      if (bucketName == null || bucketName.isEmpty()) {
```
Comment (seanzhougoogle) [inline src/main/java/io/cdap/delta/datastream/DatastreamDeltaSource.java]: nit: trim first before check for empty ?

File: src/main/java/io/cdap/delta/datastream/util/Utils.java [hunk 2]
```diff
@@ -822,17 +822,18 @@ public static DiscoverConnectionProfileResponse discoverConnectionProfile(Datast
...
+  public static boolean createBucketIfNotExisting(Storage storage, String bucketName, String location) throws IOException {
     // create corresponding GCS bucket
     try {
       Failsafe.with(createRetryPolicy()
                       .abortOn(t -> t instanceof StorageException && (
                           ((StorageException) t).getCode() == GCS_ERROR_CODE_CONFLICT ||
                           ((StorageException) t).getCode() == GCS_ERROR_CODE_INVALID_REQUEST)
                       ))
-        .run(() -> storage.create(BucketInfo.newBuilder(bucketName).build()));
+        .run(() -> storage.create(BucketInfo.newBuilder(bucketName).setLocation(location).build()));
```
Comment (seanzhougoogle) [inline src/main/java/io/cdap/delta/datastream/util/Utils.java]: what's the GCS behavior when location is null or empty or whitespace strings ?

**Review by seanzhougoogle — APPROVED** [review]
LGTM with minor comments

File: src/main/java/io/cdap/delta/datastream/util/Utils.java (hunk 2 above)
Comment (sumitjnn) [reply src/main/java/io/cdap/delta/datastream/util/Utils.java]: It throws 400 error for which we are failing (not retrying indefinitely after #55). Added handling for null/empty location.

File: src/main/java/io/cdap/delta/datastream/DatastreamDeltaSource.java (hunk 1 above)
Comment (sumitjnn) [reply src/main/java/io/cdap/delta/datastream/DatastreamDeltaSource.java]: Done

**Pull request merged**